crates/torii/core/src/engine.rs

```diff
@@ -359,16 +363,58 @@ impl<P: Provider + Sync> Engine<P> {
         }
     }
 
-    // Process a transaction and its receipt.
+    async fn process_transaction_with_events(
+        &mut self,
+        transaction_hash: Felt,
+        // transaction: &Transaction,
+        events: &[&EmittedEvent],
+        block_number: u64,
+        block_timestamp: u64,
+    ) -> Result<()> {
+        for (event_idx, event) in events.iter().enumerate() {
+            let event_id =
+                format!("{:#064x}:{:#x}:{:#04x}", block_number, transaction_hash, event_idx);
+
+            let event = Event {
+                from_address: event.from_address,
+                keys: event.keys.clone(),
+                data: event.data.clone(),
+            };
+            Self::process_event(
+                self,
+                block_number,
+                block_timestamp,
+                &event_id,
+                &event,
+                transaction_hash,
+            )
+            .await?;
+        }
+
+        // Commented out this transaction processor because it requires an RPC call for each
+        // transaction which is slowing down the sync process by alot.
+        // Self::process_transaction(
+        //     self,
+        //     block_number,
+        //     block_timestamp,
+        //     transaction_hash,
+        //     transaction,
+        // )
+        // .await?;
+
+        Ok(())
+    }
+    // Process a transaction and events from its receipt.
     // Returns whether the transaction has a world event.
-    async fn process_transaction_and_receipt(
+    async fn process_transaction_with_receipt(
         &mut self,
         transaction_hash: Felt,
-        transaction: &Transaction,
+        // transaction: &Transaction,
```

Comment:
Maybe we should just remove all of these for now. We can add them back when if we uncomment the transaction processor